app/src/main/java/org/mozilla/focus/popup/PopupFactory.java

```diff
@@ -0,0 +1,96 @@
+package org.mozilla.focus.popup;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.URLUtil;
+import android.widget.ImageView;
+import android.widget.PopupWindow;
+import android.widget.TextView;
+
+import org.mozilla.focus.R;
+import org.mozilla.focus.session.Session;
+import org.mozilla.focus.session.SessionManager;
+import org.mozilla.focus.utils.DrawableUtils;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class PopupFactory {
```

Comment:
\[PMD\] INFO: The class 'PopupFactory' has a Cyclomatic Complexity of 11 \(Highest = 10\)\.